packages/migrate/src/commands/DbPull.ts

```diff
@@ -264,11 +270,13 @@ Then you can run ${chalk.green(getCommandWithExecutor('prisma db pull'))} again.
 
         // TODO: this error is misleading, as it gets thrown even when the schema is valid but the protocol of the given
         // '--url' argument is different than the one written in the schema.prisma file.
+        // We should throw another error earlier in case the URL protocol is not compatible with the schema provider.
```

Comment:
I had spotted this error handling bug (https://github.com/prisma/prisma/issues/13076) over 6 months ago. I have left a couple of other TODOs in the file, as we should be able to throw an error before this code block in case the protocol of the --url input is not compatible with the provider attribute in the prisma.schema file.